service/effector/src/effector.rs

```diff
@@ -97,7 +103,32 @@ pub fn put(file_path: String, api_multiaddr: String, timeout_sec: u64) -> IpfsPu
     run_ipfs(cmd).map(|res| res.trim().to_string()).into()
 }
 
-/// Get file by provided hash from IPFS, save it to a `file_path, and return that path
+/// Put dag from specified path to IPFS and return its hash.
+#[marine]
+pub fn dag_put(file_path: String, api_multiaddr: String, timeout_sec: u64) -> IpfsPutResult {
+    log::info!("dag_put called with file path {}", file_path);
+
+    if !std::path::Path::new(&file_path).exists() {
+        return IpfsPutResult {
+            success: false,
+            error: format!("path {} doesn't exist", file_path),
+            hash: "".to_string(),
+        };
+    }
+
+    let args = vec![
+        String::from("dag"),
+        String::from("put"),
+        inject_vault_host_path(file_path),
+    ];
+    let cmd = make_cmd_args(args, api_multiaddr, timeout_sec);
+
+    log::info!("ipfs dag put args {:?}", cmd);
```

Comment:
sure you want to print that on every call?

Reply:
Nope, I will remove